fxa-rust-client/src/scoped_keys.rs

```diff
@@ -0,0 +1,103 @@
+use errors::*;
+
+use base64;
+use byteorder::{ByteOrder, BigEndian};
+use ring;
+use ring::{aead, agreement, rand};
+use ring::agreement::EphemeralPrivateKey;
+use serde_json;
+use sha2::{Sha256, Digest};
+use untrusted::Input;
+
+pub struct ScopedKeysFlow {
+  private_key: EphemeralPrivateKey
+}
+
+impl ScopedKeysFlow {
+  pub fn with_random_ecdh_key() -> Result<ScopedKeysFlow> {
+    let rng = rand::SystemRandom::new();
+    let private_key = EphemeralPrivateKey::generate(&agreement::ECDH_P256, &rng)?;
+    Ok(ScopedKeysFlow {
+      private_key
+    })
+  }
+
+  pub fn generate_keys_jwk(&self) -> Result<String> {
+    let mut pub_key = vec![0u8; self.private_key.public_key_len()];
+    &self.private_key.compute_public_key(&mut pub_key)?;
+    // Uncompressed representation:
+    // First byte is 4, then 32 bytes for x, and 32 bytes for y.
```

Comment:
It would be useful to link to external documentation on the format you're encoding into here, for reference.